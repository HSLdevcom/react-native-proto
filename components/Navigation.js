import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {Actions} from 'react-native-router-flux';

const styles = StyleSheet.create({
    container: {
        backgroundColor: 'rgb(255, 255, 255)',
        width: '100%',
    },
    text: {
        color: 'rgb(17, 17, 17)',
        textAlign: 'center',
    },
});

function Navigation() {
    return (
        <View style={styles.container}>
            <Text style={styles.text} onPress={Actions.main}>Main</Text>
            <Text style={styles.text} onPress={Actions.test}>Test</Text>
        </View>
    );
}

export default Navigation;
