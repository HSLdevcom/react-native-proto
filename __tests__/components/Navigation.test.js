// https://facebook.github.io/jest/blog/2016/07/27/jest-14.html
import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Navigation from '../../app/components/Navigation';

it('renders correctly', () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
});
