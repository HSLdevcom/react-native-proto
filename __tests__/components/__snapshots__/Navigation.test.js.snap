// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}
    style={
      Object {
        "color": "rgb(17, 17, 17)",
        "textAlign": "center",
      }
    }
  >
    Main
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}
    style={
      Object {
        "color": "rgb(17, 17, 17)",
        "textAlign": "center",
      }
    }
  >
    Test
  </Text>
</View>
`;
